docs/using: be more specific about how hooks work (#1284)
rauc/rauc: docs/using.rst

Bundle-Based Customization: Hooks
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

Unlike handlers, hooks are defined in the update bundle and must be
specified in the bundle's :ref:`sec_ref_manifest` file.

> All hooks are handled by a common executable that should also be part of the
> bundle.

Reviewer: ```suggestion
All hooks are handled by a common executable that must be included in the
bundle.
```

Author: I actually assumed that it would be possible to have absolute paths here. But in fact it isn't really because of the indirect check from https://github.com/rauc/rauc/commit/c3b7c7d64e0891b9712e756fe092fd97f59a2fa0.

Unlike handlers, hooks are part of the update bundle and must be
specified in the bundle's :ref:`sec_ref_manifest` file and handled by a common

> executable that should be part of the bundle.

Reviewer: `part of the (update) bundle` is now at the beginning and end of the sentence, which sounds strange.

  [hooks]
  filename=hook

The ``filename`` must match the name of the script or binary executable placed
inside the content folder the bundle is generated from.

> The actual hook invocation must be registered explicitly in the manifest via

Reviewer: ```suggestion
The actual hook invocation must be registered for each `[image.*]` in the manifest via
```

Author: I had this first, but this would exclude "bundle hooks" (i.e. install-check) which are not inside an `image` section. So the other alternative would be to mention both. `[image.*]` and `[hooks]`.
